arch/arc/Kconfig

```diff
@@ -26,6 +26,23 @@ config CPU_ARCHS
 	  This option signifies the use of an ARC HS CPU
 
 
+choice
+	prompt "ARC Instruction Set"
+	default ISA_ARCV2
+
+config ISA_ARCV2
+	bool "ARC ISA v2"
+	select ARCH_HAS_STACK_PROTECTION if ARC_HAS_STACK_CHECKING || (ARC_MPU && ARC_MPU_VER !=2)
+	select ARCH_HAS_USERSPACE if ARC_MPU
```

Comment:
I understand that it's a copy-paste of existing code, still to me it looks a bit strange why we force select `ARCH_HAS_USERSPACE` just because there's an MPU (Memory Protection Unit). I think for userspace we need HW configuration which supports kernel/user-modes, which in case of ARC might be SecureShield etc, no?

Reply:
All ARC CPUs support user/kernel modes (that's unrelated to the Secure mode SecureShield adds).
In Zephyr, userspace support requires MPU (or MMU protections), so I think this if ARC_MPU is fine.